helper/testhelpers/testhelpers.go

```diff
@@ -262,215 +271,273 @@ func PassthroughWithLocalPathsFactory(ctx context.Context, c *logical.BackendCon
 }
 
 func ConfClusterAndCore(t testing.T, conf *vault.CoreConfig, opts *vault.TestClusterOptions) (*vault.TestCluster, *vault.TestClusterCore) {
-	if conf.Physical != nil || conf.HAPhysical != nil {
-		t.Fatalf("conf.Physical and conf.HAPhysical cannot be specified")
-	}
-	if opts.Logger == nil {
-		t.Fatalf("opts.Logger must be specified")
-	}
+	cluster := MakeClusterAndStorage(t, InmemBackendSetup, conf, opts)
+	return cluster, cluster.Cores[0]
+}
 
-	inm, err := inmem.NewTransactionalInmem(nil, opts.Logger)
-	if err != nil {
-		t.Fatal(err)
+func MakeClusterAndStorage(t testing.T, setup *BackendSetup, conf *vault.CoreConfig, opts *vault.TestClusterOptions) *vault.TestCluster {
+	{
+		var coreConfig vault.CoreConfig
+		if conf != nil {
+			if conf.Physical != nil || conf.HAPhysical != nil {
+				t.Fatalf("conf.Physical and conf.HAPhysical cannot be specified")
+			}
+			coreConfig = *conf
+		}
+		conf = &coreConfig
 	}
-	inmha, err := inmem.NewInmemHA(nil, opts.Logger)
-	if err != nil {
-		t.Fatal(err)
+	if setup == nil {
+		setup = InmemBackendSetup
 	}
 
-	coreConfig := *conf
-	coreConfig.Physical = inm
-	coreConfig.HAPhysical = inmha.(physical.HABackend)
-	coreConfig.CredentialBackends = map[string]logical.Factory{
+	if setup.DisablePerformanceStandby {
+		conf.DisablePerformanceStandby = true
+	}
+	conf.CredentialBackends = map[string]logical.Factory{
 		"approle":  credAppRole.Factory,
 		"userpass": credUserpass.Factory,
+		"ldap":     ldap.Factory,
 	}
-	vault.AddNoopAudit(&coreConfig)
-	cluster := vault.NewTestCluster(t, &coreConfig, opts)
-	cluster.Start()
 
-	cores := cluster.Cores
-	core := cores[0]
+	opts = getClusterDefaultsOpts(t, opts, "")
+	if setup.KeepStandbysSealed {
+		opts.KeepStandbysSealed = true
+	}
+	opts.SetupFunc = setup.Setup
+	opts.CleanupFunc = setup.Cleanup
```

Comment:
Why introduce a third object that has the same fields as the opts struct? Seems like we can already pass the intentions in via TestClusterOptions

Reply:
Maybe the setup strategy could just mutate the opts object before calling `MakeClusterAndStorage` something like:

`MakeClusterAndStorage(t, conf, InmemBackendSetup{}.Configure(opts))`

Reply:
I followed your mutator suggestion.  The reason I added the new object was because it seemed less confusing, but in hindsight I'm not sure it actually was.  I think I'm happier with your approach.